src/compiler/program.ts

```diff
@@ -673,8 +851,8 @@ namespace ts {
             return getBaseFileName(fileName).indexOf(".") >= 0;
         }
 
-        function processRootFile(fileName: string, isDefaultLib: boolean) {
-            processSourceFile(normalizePath(fileName), isDefaultLib);
+        function processRootFile(fileName: string, isDefaultLib: boolean, supportedExtensions: string[]) {

```

Comment:
i do not see the reason for why these are propagated here. the module resolver use their own copy from calling getSupportedExtensions